modules/ngx_tcp_generic_proxy_module.c

```diff
@@ -233,7 +233,42 @@ ngx_tcp_proxy_init_upstream(ngx_connection_t *c, ngx_tcp_session_t *s)
 }
 
 
-static void 
+static void
+ngx_tcp_proxy_send_proxy_protocol(ngx_tcp_session_t *s, ngx_tcp_upstream_t *u)
+{
+    ngx_connection_t         *c;
+    ngx_str_t                proxy_line;
+    struct sockaddr_in *client, *listener;
+    u_char *src_addr, *dst_addr;
+
+    c = u->peer.connection;
+    c->log->action = "ngx_tcp_proxy_send_proxy_protocol";
+
+    client = (struct sockaddr_in*)s->connection->sockaddr;
+    listener = (struct sockaddr_in*)s->connection->listening->sockaddr; // doesn't appear to be correct
```

Comment:
I gave it a go, doesn't seem to return the correct values either. IP is 0.0.0.0 and the port is not the port defined in the configuration. Not that important anyway IMO